docs/en/latest/http3.md

```diff
@@ -0,0 +1,128 @@
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+[HTTP/3](https://en.wikipedia.org/wiki/HTTP/3) is the third major version of the Hypertext Transfer Protocol (HTTP). Unlike its predecessors which rely on TCP, HTTP/3 is based on [QUIC (Quick UDP Internet Connections) protocol](https://en.wikipedia.org/wiki/QUIC). It brings several benefits that collectively result in reduced latency and improved performance:
+
+ * enabling seamless transition between different network connections, such as switching from Wi-Fi to mobile data.
+ * eliminating head-of-line blocking, so that a lost packet does not block all streams.
+ * negotiate TLS versions at the same time as the TLS handshakes, allowing for faster connections.
+ * providing encryption by default, ensuring that all data transmitted over an HTTP/3 connection is protected and confidential.
+ * providing zero round-trip time (0-RTT) when communicating with servers that clients already established connections to.
+
+ APISIX currently supports HTTP/3 connections between downstream clients and APISIX. HTTP/3 connections with upstream services is not yet supported. Contributions are welcomed.
+
+ :::caution
+
+ This feature has not been tested at scale and therefore, is not recommended for production use.
+
+ :::
+
+## Usage example
+
+1. To allow for HTTP/3 connections on 9443 port, add the following configurations to the `config.yaml` file:
+
+```yaml
+apisix:
+  ssl:
+    listen:
+      - port: 9443
+        enable_http3: true   # enable HTTP/3
+    ssl_protocols: TLSv1.3 # enable TLSv1.3
+```
+
+**note** `enable_http3` requires `TLSv1.3`
+
+2. Create an SSL object for test.com.
+
+```shell
+curl http://127.0.0.1:9180/apisix/admin/ssls/1 \
+-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+     "cert" : "'"$(cat t/certs/apisix.crt)"'",
+     "key": "'"$(cat t/certs/apisix.key)"'",
+     "snis": ["test.com"]
+}'
+```
+
+3. Create route
+
+```shell
+curl http://127.0.0.1:9180/apisix/admin/routes/1 \
+-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -i -d '
+{
+    "uri": "/get",
+    "hosts": ["test.com"],
+    "methods": ["GET"],
+    "upstream": {
+        "type": "roundrobin",
+        "nodes": {
+            "httpbin.org": 1
+        }
+    }
+}'
+```
+
+4. Access verification
+
+Access test.com using HTTP/3:
+
+- curl version 7.88.0+, requires [http-only](https://github.com/curl/curl/blob/master/docs/cmdline-opts/http3-only.md) option.
```

Comment:
`8.4.0` also doesn't seem to support this option

![image](https://github.com/apache/apisix/assets/39619599/1e3eb026-61ac-438a-8421-2f52d89bda17)